### PR TITLE
Update build documentation

### PR DESCRIPTION
**Pull request by ofek**
Also documented the cross compilation temporary limitation https://github.com/cross-rs/cross/issues/1215

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,14 +50,32 @@ For a more streamlined workflow, consider using the built-in [app](https://hatch
...
 
+### Local repository
+
+Clone this repository then enter the cloned directory and run:
+
+```
+cargo build --release
+```
+
+The executable will be located at `target/release/pyapp.exe` if on Windows or `target/release/pyapp` otherwise. If a particular [target](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget) has been set then the `release` directory will be nested one level deeper under `target/<TARGET>`.
```
Comment (Emilgardis) [inline README.md:75]: This is not correct, when cargo is invoked with `--target` it unconditionally puts it in `target/<triplet/...`, and cross will always specify the target even if you call `cross build` without a specific target.

Comment (ofek) [reply README.md:75]: So I should change it to say "if a target has been set or if cross is used"?

Comment (Emilgardis) [reply README.md:75]: Ah right, you should be getting a warning though if your default on windows is msvc

```
[cross] warning: `cross` does not provide a Docker image for target x86_64-pc-windows-msvc, specify a custom image in `Cross.toml`.
[cross] note: Falling back to `cargo` on the host.
```

Comment (ofek) [reply README.md:75]: Fixed, thank you! https://github.com/ofek/pyapp/commit/1384865c7bb03d9d4fdd598ce7aca5a57c73bd63

**Pull request merged**